### DISCUSSION TITLE
p.244 リスト20.24最後の行のr.Context()について

### DISCUSSION THREAD

**Discussion by youta32449999** [budougumi0617/go_todo_app#57] [Ideas]
p.244 リスト20.24の最後の行が下記のようになっております。
プログラムの意味は変わりませんが、他のハンドラ同様`r.Context()`ではなく`ctx`を使用するのが良いと思いました。
(あえてここだけ`ctx`を使用しない理由があるのかもしれないと深読みして読み進めるのが止まってしまうため)
```go
// handler/login.go

RespondJSON(r.Context(), w, rsp, http.StatusOK)
```

**Comment by budougumi0617** [comment]
ご指摘のとおりで、処理の途中で `ctx` 変数が更新されている可能性もあるので `r.Context()` ではなく `ctx` 変数を使うべきでした。
正誤表とコードを修正いたしました。ありがとうございます。

https://github.com/budougumi0617/go_todo_app/releases/tag/v1.0.7
